atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -15,183 +15,69 @@
-                        "current_max_ts",
-                        ColumnValueDescription.forType(ValueType.FIXED_LONG)))),
-            ConflictHandler.IGNORE_ALL);
-
     private static final long INITIAL_VALUE = 10000L;
 
     @GuardedBy("this")
     private long currentLimit = -1;
 
-    private final CassandraClientPool clientPool;
+    private final CassandraTimestampStore cassandraTimestampStore;
 
     public static TimestampBoundStore create(CassandraKeyValueService kvs) {
-        kvs.createTable(AtlasDbConstants.TIMESTAMP_TABLE, TIMESTAMP_TABLE_METADATA.persistToBytes());
-        return new CassandraTimestampBoundStore(kvs.clientPool);
+        CassandraTimestampStore cassandraTimestampStore = new CassandraTimestampStore(kvs);
+        cassandraTimestampStore.createTimestampTable();
+        return new CassandraTimestampBoundStore(cassandraTimestampStore);
     }
 
-    private CassandraTimestampBoundStore(CassandraClientPool clientPool) {
+    private CassandraTimestampBoundStore(CassandraTimestampStore cassandraTimestampStore) {
         DebugLogger.logger.info(
                 "Creating CassandraTimestampBoundStore object on thread {}. This should only happen once.",
                 Thread.currentThread().getName());
-        this.clientPool = Preconditions.checkNotNull(clientPool, "clientPool cannot be null");
+        this.cassandraTimestampStore = cassandraTimestampStore;
     }
 
     @Override
     public synchronized long getUpperLimit() {
         DebugLogger.logger.debug("[GET] Getting upper limit");
-        return clientPool.runWithRetry(new FunctionCheckedException<Client, Long, RuntimeException>() {
-            @Override
-            public Long apply(Client client) {
-                ByteBuffer rowName = getRowName();
-                ColumnPath columnPath = new ColumnPath(AtlasDbConstants.TIMESTAMP_TABLE.getQualifiedName());
-                columnPath.setColumn(getColumnName());
-                ColumnOrSuperColumn result;
-                try {
-                    result = client.get(rowName, columnPath, ConsistencyLevel.LOCAL_QUORUM);
-                } catch (NotFoundException e) {
-                    result = null;
-                } catch (Exception e) {
-                    throw Throwables.throwUncheckedException(e);
-                }
-                if (result == null) {
-                    DebugLogger.logger.info("[GET] Null result, setting timestamp limit to {}", INITIAL_VALUE);
-                    cas(client, null, INITIAL_VALUE);
-                    return INITIAL_VALUE;
-                }
-                Column column = result.getColumn();
-                currentLimit = PtBytes.toLong(column.getValue());
-                DebugLogger.logger.info("[GET] Setting cached timestamp limit to {}.", currentLimit);
-                return currentLimit;
-            }
-        });
+        Optional<Long> currentBound = cassandraTimestampStore.getUpperLimit();
+        long boundToStore;
```

Comment:
OK. We can actually get rid of `boundToStore` altogether after the method extraction.